scripts/linux/buildEdgeProxy.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+# This script copies the edge-proxy executable files that goes into the azureiotedge-proxy image,
+# for each supported arch (x86_64, arm32v7, arm64v8).
+# It then publishes executable files along with their corresponding dockerfiles to the publish directory,
+# so that buildImage.sh can build the container image.
+
+set -e
+
+###############################################################################
+# Define Environment Variables
+###############################################################################
+ARCH=$(uname -m)
+SCRIPT_NAME=$(basename "$0")
+PROJECT=
+DOCKERFILE=
+DOCKER_IMAGENAME=
+DEFAULT_DOCKER_NAMESPACE="microsoft"
+DOCKER_NAMESPACE=${DEFAULT_DOCKER_NAMESPACE}
+BUILD_BINARIESDIRECTORY=${BUILD_BINARIESDIRECTORY:-$BUILD_REPOSITORY_LOCALPATH}
+PUBLISH_DIR=${BUILD_BINARIESDIRECTORY}/publish
+BUILD_CONFIGURATION="release"
```

Comment:
Shellcheck warns about this variable:
```
In buildEdgeProxy.sh line 97:
            BUILD_CONFIGURATION="$arg"
            ^-- SC2034: BUILD_CONFIGURATION appears unused. Verify it or export it.
```

Reply:
This variable provided as ENV by build agent

Reply:
Sure, but do we need it if its not being used in the script?

Reply:
While this variable is not being used now, we need to process the `--configuration` argument in this script since this script is being used from the `build-edgelet-linux.yaml` template to build both the proxy image and the diagnostics image and for the latter this variable _is_ used. We could solve this by having the processing for `--configuration` parameter in `buildEdgeProxy.sh` that just ignores the param. Feel free to leave it be though.